_posts/2024-05-22-developing-openwrt-in-qemu.adoc

```diff
@@ -0,0 +1,230 @@
+---
+title: Developing OpenWRT in QEMU with Eclipse Che
+layout: post
+author: Oleksii Orel
+description: >-
+  Developing OpenWRT helloworld package in QEMU machine emulator and virtualizer with Eclipse Che.
+categories: []
+keywords: ['openwrt', 'quemu', 'gdb', 'developing ', 'workspace']
+slug: /@olexii.orel/openwrt-helloworld-package
+---
+
+https://openwrt.org/[OpenWrt] is a highly extensible GNU/Linux distribution for embedded devices (typically wireless routers). OpenWrt is built from the ground up to be a full-featured, easily modifiable operating system for embedded devices. You can have all the features you need with no bloat, powered by a modern Linux kernel.
+
+We can create and develop https://openwrt.org/docs/guide-developer/creating-a-cmake-package-in-openwrt[OpenWRT CMake packages] with Eclipse Che.
+
+In this blog post, we will review developing OpenWRT in https://www.qemu.org/[QEMU] with Eclipse Che in 4 parts:
+
+<<part1>>
+
+<<part2>>
+
+<<part3>>
+
+<<part4>>
+
+
+NOTE: In this blog post, we will use the next devfile:
+[source,yaml]
```

Comment:
I do not think it is the right place for devfile - let use URL to the repository in the note and provide instructions for running workspace on developer sandbox via https://workspaces.openshift.com#https://github.com/che-incubator/openwrt-helloworld-package

Reply:
We can add the devfile with description of CDE (Cloud Development Environment) in part 1

Reply:
I have fixed it.